### PR TITLE
terraform-providers: new providers + version bumps

### PR DESCRIPTION
**Pull request by stephengroat**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/cluster/terraform-providers/providers.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #  <organisation>/<repo>                                         - include only the named repository.
 
 # include all terraform-providers
-terraform-providers terraform-provider- terraform-provider-\\(azure-classic\\|scaffolding\\|google-beta\\|skytap\\)
+terraform-providers terraform-provider- terraform-provider-\\(azure-classic\\|scaffolding\\)
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/terraform-providers/providers.txt]: Do you know why those where disabled in the first place?

Comment (stephengroat) [reply pkgs/applications/networking/cluster/terraform-providers/providers.txt]: from what i can tell, these providers weren't tagged with git versions for a long time (which prevented the `update-all` script from working).

I can remove these if needed, just saw that there were versioned now

Comment (zimbatm) [reply pkgs/applications/networking/cluster/terraform-providers/providers.txt]: google-beta is also desirable to have, not sure about skytap

they are pulling out all the beta features from the google provider and moving them into google-beta, but a lot of google feature stay in beta for a long time

**Review by zimbatm — CHANGES_REQUESTED** [review]
google and google-beta are colliding when building the full version of terraform:
```
$ nix-build nixpkgs/ -A terraform.full
these derivations will be built:
  /nix/store/yz2psm8bpcrq6nw7z5my2x903ih81km1-tf-plugin-env.drv
  /nix/store/hv2429i5746vwqla08fh0sh3swsdzfk5-terraform-0.11.11.drv
building '/nix/store/yz2psm8bpcrq6nw7z5my2x903ih81km1-tf-plugin-env.drv'...
collision between `/nix/store/z673jw5bflk84chhd90iiys6w8d7pl39-terraform-provider-google-beta-1.20.0-bin/bin/scripts' and `/nix/store/g6f2qc9pp7cqk5q53sl5by4hs83dlmzn-terraform-provider-google-1.20.0-bin/bin/scripts'
builder for '/nix/store/yz2psm8bpcrq6nw7z5my2x903ih81km1-tf-plugin-env.drv' failed with exit code 25
cannot build derivation '/nix/store/hv2429i5746vwqla08fh0sh3swsdzfk5-terraform-0.11.11.drv': 1 dependencies couldn't be built
error: build of '/nix/store/hv2429i5746vwqla08fh0sh3swsdzfk5-terraform-0.11.11.drv' failed
```

**Comment by zimbatm** [comment]
```
diff --git a/pkgs/applications/networking/cluster/terraform-providers/default.nix b/pkgs/applications/networking/cluster/terraform-providers/default.nix
index 040e5161bc1..702a7cb7eb1 100644
--- a/pkgs/applications/networking/cluster/terraform-providers/default.nix
+++ b/pkgs/applications/networking/cluster/terraform-providers/default.nix
@@ -11,7 +11,6 @@ let
       inherit (data) owner repo version sha256;
       name = "${repo}-${version}";
       goPackagePath = "github.com/${owner}/${repo}";
+      subPackages = [ "." ];
       src = fetchFromGitHub {
         inherit owner repo sha256;
         rev = "v${version}";
```

**Review by zimbatm — APPROVED** [review]

**Comment by stephengroat** [comment]
@zimbatm thanks for the help!

**Pull request merged**